.github/workflows/prerelease.yml

```diff
@@ -0,0 +1,468 @@
+name: prerelease
+
+on:
+  pull_request:
+    paths:
+    - '.github/workflows/prerelease.yml'
+  schedule:
+    # Run every day at 11pm (PST) - cron uses UTC times
+    - cron:  '0 7 * * *'
+
+jobs:
+  buildup_SpecsReleasing_repo:
+    # Don't run on private repo unless it is a PR.
+    if: github.repository != 'FirebasePrivate/firebase-ios-sdk' || github.event_name == 'pull_request'
+    env:
+      bot_token_secret: ${{ secrets.GHASecretsGPGPassphrase1 }}
+      local_repo: specstesting
+      local_sdk_repo_dir: /tmp/test/firebase-ios-sdk
+      nightly_version: 6.33.0
+      podspec_repo_branch: release-6.33.0
+    runs-on: macOS-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Get token
+      run: |
+         scripts/decrypt_gha_secret.sh scripts/gha-encrypted/oss-bot-access.txt.gpg \
+         oss-bot-access.txt "$bot_token_secret"
+         scripts/decrypt_gha_secret.sh scripts/gha-encrypted/bot-access.txt.gpg \
+         bot-access.txt "$bot_token_secret"
+    - name: Update SpecsReleasing repo setup
+      run: |
+         ossbotaccess=`cat oss-bot-access.txt`
+         BOT_TOKEN="${ossbotaccess}" test_version="${nightly_version}" sdk_version_config="${GITHUB_WORKSPACE}/scripts/create_spec_repo/RC_firebase_sdk.textproto" scripts/release_testing_setup.sh RC_testing
+    - name: Update SpecsReleasing repo
+      run: |
+        botaccess=`cat bot-access.txt`
+        cd scripts/create_spec_repo/
+        swift build
+        pod repo add --silent "${local_repo}" https://"$botaccess"@github.com/FirebasePrivate/SpecsReleasing.git
+        BOT_TOKEN="${botaccess}"  .build/debug/spec-repo-builder --sdk-repo "${local_sdk_repo_dir}" --local-spec-repo-name "${local_repo}" --sdk-repo-name SpecsReleasing --pod-sources "https://${botaccess}@github.com/FirebasePrivate/SpecsReleasing" --pod-sources "https://cdn.cocoapods.org/"
+    - name: Clean Artifacts
+      if: ${{ always() }}
+      run: |
+        pod repo remove "${local_repo}"
+        rm -rf bot-access.txt
+  remoteconfig_quickstart:
+    # Don't run on private repo unless it is a PR.
```

Comment:
It would require a few special case tests, but wondering if we should to all the jobs via one matrixed job?

Reply:
I made some experiments and it turns out  we might need to update scripts to have matrix. E.g. Crashlytics need to have two binaries,run and upload-symbols. Also, some tests have objc while some do not have. I think we might submit this pr to run prerelease test first, and another pr to refactor nightly release and prerelease workflows.

Reply:
SGTM